cmd/promtool/main.go

```diff
@@ -303,9 +304,74 @@ func checkRules(filename string) (int, []error) {
 		numRules += len(rg.Rules)
 	}
 
+	showWarning := false
+	dLabels, dMetrics, nLabels, nMetrics := checkDuplicates(rgs.Groups)
+	if dLabels != 0 {
+		showWarning = true
+		fmt.Printf("%d duplicate label(s) found.\n", dLabels)
+		for _, n := range nLabels {
+			fmt.Println(n)
+		}
+	}
+	if dMetrics != 0 {
+		showWarning = true
+		fmt.Printf("%d duplicate metric(s) found.\n", dMetrics)
+		for _, n := range nMetrics {
+			fmt.Println(n)
+		}
+	}
+	if showWarning {
+		fmt.Println("Might cause inconsistency while recording expressions.")
+	}
+
 	return numRules, nil
 }
 
+func checkDuplicates(r []rulefmt.RuleGroup) (int, int, []string, []string) {
+	duplicateLabels, duplicateMetrics := 0, 0
+	labelNames, metricName := []string{}, []string{}
+	for _, rule := range r {
+		var stackLabels []map[string]string
+		var stackMetrics []string
+		for _, props := range rule.Rules {
+			for _, labels := range stackLabels {
```

Comment:
A map would be more efficient, and easier to understant

Reply:
sure